subsys/net/openthread/rpc/client/ot_rpc_diag_client.c

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <ot_rpc_ids.h>
+
+#include <nrf_rpc_cbor.h>
+
+#include <openthread/error.h>
+#include <openthread/dataset.h>
+#include <openthread/instance.h>
+#include <openthread/link.h>
+#include <openthread/thread.h>
+#include <zephyr/logging/log.h>
+
+NRF_RPC_GROUP_DECLARE(ot_group);
+LOG_MODULE_DECLARE(ot_rpc, LOG_LEVEL_DBG);
+
+static char version[256];
+static otExtAddress mac = {.m8 = {0xF4, 0xCE, 0x36, 0x24, 0x95, 0x6A, 0xD1, 0xD0}};
+static otExtendedPanId ext_pan_id;
+static char thread_network_name[OT_NETWORK_NAME_MAX_SIZE + 1];
+static otMeshLocalPrefix mesh_prefix;
+
+static void get_uint_t(int id, void *result, size_t result_size)
+{
+	const size_t cbor_buffer_size = 0;
+	struct nrf_rpc_cbor_ctx ctx;
+
+	NRF_RPC_CBOR_ALLOC(&ot_group, ctx, cbor_buffer_size);
+
+	nrf_rpc_cbor_cmd_rsp_no_err(&ot_group, id, &ctx);
+
+	if (!zcbor_uint_decode(ctx.zs, result, result_size)) {
+		nrf_rpc_cbor_decoding_done(&ot_group, &ctx);
+		ot_rpc_report_decoding_error(id);
+		return;
+	}
+
+	nrf_rpc_cbor_decoding_done(&ot_group, &ctx);
+}
+
+static int get_string(int id, char *buffer, size_t buffer_size)
+{
+	const size_t cbor_buffer_size = 0;
+	size_t bytes_copied = 0;
+	struct zcbor_string zst;
+	struct nrf_rpc_cbor_ctx ctx;
+
+	NRF_RPC_CBOR_ALLOC(&ot_group, ctx, cbor_buffer_size);
+
+	nrf_rpc_cbor_cmd_rsp_no_err(&ot_group, id, &ctx);
+
+	if (!zcbor_bstr_decode(ctx.zs, &zst)) {
+		nrf_rpc_cbor_decoding_done(&ot_group, &ctx);
+		ot_rpc_report_decoding_error(id);
+		goto exit;
+	}
+
+	if (buffer_size >= zst.len) {
+		bytes_copied = zst.len;
+	} else {
+		bytes_copied = 0;
+		ot_rpc_report_decoding_error(id);
```

Comment:
`_decoding_done` is not called in this case. Call it before checking the length?